src/number_theory/lucas_lehmer.lean

```diff
@@ -0,0 +1,558 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Mario Carneiro, Scott Morrison, Ainsley Pahljina
+-/
+
+import data.nat.parity
+import data.zmod.basic
+import group_theory.order_of_element
+import ring_theory.fintype
+
+/-!
+# The Lucas-Lehmer test for Mersenne primes.
+
+We define `Lucas_Lehmer_residue : Π p : ℕ, zmod (2^p - 1)`, and
```

Comment:
Minor issue: We have things like `legendre_sym` all lower case. Should this be `lucas_lehmer_residue`?
Idem for `Mersenne` vs `mersenne`...

Reply:
All lower case now.